runtime/jcl/common/criusupport.c

```diff
@@ -39,38 +39,79 @@ setupJNIFieldIDs(JNIEnv *env)
-		Assert_JCL_notNull(vm->criuResultInit);
-	} else {
-		vmFuncs->internalEnterVMFromJNI(currentThread);
+	criuGlobals = (J9CRIUGlobals *)j9mem_allocate_memory(sizeof(J9CRIUGlobals), OMRMEM_CATEGORY_VM);
+	if (NULL == criuGlobals) {
 		vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
-		vmFuncs->internalExitVMToJNI(currentThread);
+		goto done;
 	}
+	memset(criuGlobals, 0, sizeof(J9CRIUGlobals));
+
+	if (NULL == criuGlobals->criuResultTypeClass
+		&& NULL == criuGlobals->criuResultClass
+		&& NULL == criuGlobals->criuSupportClass
+	) {
+		criuResultTypeClass = (*env)->FindClass(env, "org/eclipse/openj9/criu/CRIUSupport$CRIUResultType");
+		Assert_JCL_notNull(criuResultTypeClass);
+		criuGlobals->criuResultTypeClass = (*env)->NewGlobalRef(env, criuResultTypeClass);
+
+		criuResultClass = (*env)->FindClass(env, "org/eclipse/openj9/criu/CRIUSupport$CRIUResult");
+		Assert_JCL_notNull(criuResultClass);
+		criuGlobals->criuResultClass = (*env)->NewGlobalRef(env, criuResultClass);
+
+		criuSupportClass = (*env)->FindClass(env, "org/eclipse/openj9/criu/CRIUSupport");
+		Assert_JCL_notNull(criuSupportClass);
+		criuGlobals->criuSupportClass = (*env)->NewGlobalRef(env, criuSupportClass);
+
+		if (NULL != criuGlobals->criuResultTypeClass
+			&& NULL != criuGlobals->criuResultClass
+			&& NULL != criuGlobals->criuSupportClass
+		) {
+			criuGlobals->criuSupportSuccess = (*env)->GetStaticFieldID(env, criuGlobals->criuResultTypeClass, "SUCCESS", "Lorg/eclipse/openj9/criu/CRIUSupport$CRIUResultType;");
+			Assert_JCL_notNull(criuGlobals->criuSupportSuccess);
+			criuGlobals->criuSupportUnsupportedOperation  = (*env)->GetStaticFieldID(env, criuGlobals->criuResultTypeClass, "UNSUPPORTED_OPERATION", "Lorg/eclipse/openj9/criu/CRIUSupport$CRIUResultType;");
+			Assert_JCL_notNull(criuGlobals->criuSupportUnsupportedOperation);
+			criuGlobals->criuSupportInvalidArguments = (*env)->GetStaticFieldID(env, criuGlobals->criuResultTypeClass, "INVALID_ARGUMENTS", "Lorg/eclipse/openj9/criu/CRIUSupport$CRIUResultType;");
+			Assert_JCL_notNull(criuGlobals->criuSupportInvalidArguments);
+			criuGlobals->criuSupportSystemCheckpointFailure = (*env)->GetStaticFieldID(env, criuGlobals->criuResultTypeClass, "SYSTEM_CHECKPOINT_FAILURE", "Lorg/eclipse/openj9/criu/CRIUSupport$CRIUResultType;");
+			Assert_JCL_notNull(criuGlobals->criuSupportSystemCheckpointFailure);
+			criuGlobals->criuSupportJVMCheckpointFailure = (*env)->GetStaticFieldID(env, criuGlobals->criuResultTypeClass, "JVM_CHECKPOINT_FAILURE", "Lorg/eclipse/openj9/criu/CRIUSupport$CRIUResultType;");
+			Assert_JCL_notNull(criuGlobals->criuSupportJVMCheckpointFailure);
+			criuGlobals->criuSupportJVMRestoreFailure = (*env)->GetStaticFieldID(env, criuGlobals->criuResultTypeClass, "JVM_RESTORE_FAILURE", "Lorg/eclipse/openj9/criu/CRIUSupport$CRIUResultType;");
+			Assert_JCL_notNull(criuGlobals->criuSupportJVMRestoreFailure);
```

Comment:
I suggest it would be more efficient to use a `jint` to distinguish the various `CRIUResultType` values. Those values can be defined in the java source and made available via (the equivalent of) `javah` in native code.

Reply:
I think we will be getting rid of `CRIUResultType` soon (#13137)